google/ggrc-core
# GGRC-7445 [Part 1] Clean up / simplify "modals_controller" controller

# Issue description

In the scope of this PR, modals_controller was cleaned up and simplified.

# Steps to test the changes

\# 1
1. Open Create/Edit modal for Task group without HR integration.
2. Check that mandatory field "Assignee" is working as needed.

\# 2
1. Open Create/Edit modal for Cycle task without HR integration.
2. Check that "Task Assignee" "Task Secondary Assignee" "Workflow" "Task Group" fields are working as needed.

\# 3
1. Open Create/Edit modal for Audit without HR integration.
2. Check that a user is able to define program, audit title, Audit Firm.

\# 4
1. Open Create/Edit modal for any object which contains autocomplete
2. Check that it works.

\# 5
1. Open Create/Edit modal for GCA.
2. "Attribute types" dropdown should work. When you type something in the input for type "Dropdown/Multiselect" and then change attribute type and return previous attribute type back again, input with typed values should be cleared.

\# 6
1. Open Mapping filter on export page
2. It should work, Create button at the end of the list shouldn't be shown.

\# 7
1. Create any object via Create modal with autocompletes, for example, Cycle task via **"Save & Add Another" button**.
2. Check that newly showed Create modal and it's autocomplete works as needed.

# Solution description

1. Most of autocomplete-related logic was moved out of modals_controller.
2. Some pieces of dead code were removed.
3. Several bugs were fixed:
     When "Task Group assignee" is removed, we had ability to save Task Group - error was shown.
     User's input was not cleared if we change type of GCA from multichoicable to non-multichoicable.

More details in comments.

# Sanity checklist




# PR Review checklist
